InteractionBase/src/commands/CAddNodeToViewByName.cpp

```diff
@@ -40,18 +41,42 @@ bool CAddNodeToViewByName::canInterpret(Visualization::Item*, Visualization::Ite
 	return commandTokens.size() >= 1 && commandTokens.first() == name();
 }
 
-CommandResult* CAddNodeToViewByName::execute(Visualization::Item*, Visualization::Item* target,
-		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>&)
+CommandResult* CAddNodeToViewByName::execute(Visualization::Item* source, Visualization::Item* target,
+		const QStringList& commandTokens, const std::unique_ptr<Visualization::Cursor>& cursor)
 {
 	if (commandTokens.size() < 2)
 		return new CommandResult(new CommandError("Please specify a node to add"));
 
+	auto currentView = target->scene()->currentViewItem();
+	QPoint posToInsert;
+	if (source == currentView && isHorizontal(cursor->region()))
```

Comment:
Doesn't the source node being a ViewItem already imply this? At least when I test it, it only does the special version of the insertion, when I actually have the cursor on the ViewItem itself. Or is checking the cursor's owner just a cleaner way of doing it?